main.cpp

```diff
@@ -6,7 +6,18 @@
 #include <unordered_map>
 #include <utility>
 #include <vector>
+#include <fstream> //for logging to file
 
+int initRand() { // get random seed
+	int seed = time(0);
+	srand(seed);
+	return seed;
+}
+
+int initRand(int seed) { // set custom seed for debugging purpouses

```

Comment:
#nice! Though now looking at this, you #should instead have just one overload and do:

```cpp
unsigned int initRand(std::optional<unsigned int> customSeed = std::nullopt)
{
    unsigned int seed = customSeed.value_or(time(nullptr));
    srand(seed);
    return seed;
}
```